### DISCUSSION TITLE
[iOS/Android] Import wallet by privateKey.

### DISCUSSION THREAD

**Discussion by remboshelby** [trustwallet/wallet-core#1812] [Q&A] [answered]
I develop application for ETH blockchain. We need to be able to import wallet by private key.

Now we are trying to create HDWallet from privateKey using method createWithData.
As I understood this method provides creation wallet using entropy.

Can you help me how to use this method with privateKey? or it is not impossible?

P.s: if in current version this functionality doesn't exist, can you add this?

**Comment by optout21** [comment, marked as answer]
HDWallet can be used to derived different coin-specific private keys from a BIP39 mnemonic.
In case you have the ETH private key, there is no need to use HDWallet.  Private key can be used directly, to derive address (via public key), or to sign transactions.
